lib/Controller/ConfigController.php

```diff
@@ -160,21 +161,20 @@ public function popupSuccessPage(string $user_name, string $user_displayname): T
 	 * @throws PreConditionNotMetException
 	 */
 	public function oauthRedirect(string $code = '', string $state = ''): RedirectResponse {
-		$configState = $this->config->getUserValue($this->userId, Application::APP_ID, 'oauth_state');
-		$clientID = $this->config->getAppValue(Application::APP_ID, 'client_id');
-		$clientSecret = $this->config->getAppValue(Application::APP_ID, 'client_secret');
+		$configState = $this->config->getUserOauthState($this->userId);
+		$clientID = $this->config->getAdminClientId();
+		$clientSecret = $this->config->getAdminClientSecret();
 
 		// anyway, reset state
-		$this->config->deleteUserValue($this->userId, Application::APP_ID, 'oauth_state');
+		$this->config->deleteUserOauthState($this->userId);
 
 		if ($clientID and $clientSecret and $configState !== '' and $configState === $state) {
-			$redirect_uri = $this->config->getUserValue($this->userId, Application::APP_ID, 'redirect_uri');
-			$adminOauthUrl = $this->config->getAppValue(Application::APP_ID, 'oauth_instance_url', Application::DEFAULT_GITLAB_URL) ?: Application::DEFAULT_GITLAB_URL;
+			$adminOauthUrl = $this->config->getAdminOauthUrl();
 			$result = $this->gitlabAPIService->requestOAuthAccessToken($adminOauthUrl, [
 				'client_id' => $clientID,
 				'client_secret' => $clientSecret,
 				'code' => $code,
-				'redirect_uri' => $redirect_uri,
+				'redirect_uri' => $this->config->getUserRedirectUri($this->userId),
```

Comment:
nitpick: would be cleaner to store the `$redirect_uri` and then use it in the function

Reply:
You mean store it in a variable first like before?

Reply:
IMO this is not necessary since the value is only used once and it is pretty clear what is going on even without the variable. With the previous way of reading config values I'd agree that it is too noisy to have it inline, but this looks fine to me.

Reply:
sure, it was just for the eyes anyway